Topics/Development_Process/NextJS_AWS_Deployment.md

```diff
@@ -0,0 +1,46 @@
+
+## Getting started with Next
+Since this is primarily a guide in the deployment process, it is assumed that the reader has their Next.js app already created and ready to go. IF that's not the case, no problem!
+
+Next.js is a react framework for the web, which enables you to create **full-stack** Web applications, all by extending React features. In addition to countless online rescources and excellent docs the Next.js team provides, getting started is even simpler with Next.js's command line interface tool, [`create-next-app`](https://nextjs.org/docs/app/api-reference/create-next-app). This helpful tool can really kickstart your development, and get your project off to a clean start.
+
+## Creating an EC2 Instance
+Now that you've hopefully got your hands a little dirty with Next.js develepment, its time to get your app off localhost and into the cloud! This process begins by creating an EC2 Instance.
+
+1) Log in to AWS, or create a free account.(Note you will need to provide credit card information, but as long as you only use free services, you won't get billed).
+2) Navigate to the **EC2 Dashboard**. ![image](https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/62919149/935781b1-e426-453c-84d6-4dec25e53901)
+3) Create a new instance by clicking **Launch Instance**.
+4) Name your instance (name doesn't really matter), select the Ubuntu server for you Amazon Machine Image, select your desired instance type (micro for free tier), create a new RSA key pair(make sure you keep track of its location). That's it for config; select the **Launch Instance** button at the bottom of the screen.
+5) Locate your instance by navigating to **All Instances**, then select your newly created instance. Open the **Security** tab, and click the security groups tab.![image](https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/62919149/bc74382e-12b9-4041-9dc3-db1dcc61a8be)
+6) Head to **Edit Inbound Rules**, and create a new rule with Type = Custom, Port range = the port your app runs on(by default, it will be 3000), Source=IPv4.
+
+Now its time to connect to your EC2 instance from your machine.
+1) Open a terminal in the same directory as the .pem key file that your downloaded earlier when creating your key pair.
+2) Click the **Connect** tab while your EC2 instance is selected ![image](https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/62919149/709744fd-9472-4ea1-a6c9-58468fb9b80e)
+3) Copy and paste bullet point #3 and the example into your terminal and run both commands. Enter `yes` when asked if you wish to continue connecting. You should now be connected to your EC2 through shell!
```

Comment:
Do you think I should add screen shot of the modal which contains these lines of code? Or copy and paste a generic template. Only problem I can foresee is that students may blindly copy and paste the template without using their proper key file.

Reply:
If this is the command to connect to the ec2 instance, then yes I think it would be helpful to include a screenshot (with your personal deployment info censored)

Reply:
Ok will do.